HW28/index.js

```diff
@@ -0,0 +1,182 @@
+class Human {
+  constructor({ name, surname, age }) {
+    this.name = name;
+    this.surname = surname;
+    this.age = age;
+  }
+
+  getFullName() {
+    return `${this.name} ${this.surname}`;
+  }
+
+  setFullName(fullName) {
+    [this.name, this.surname] = fullName.split(" ");
+  }
+}
+
+class Student extends Human {
+  constructor(info) {
+    super(info);
+    this.marks = info.marks;
+  }
+
+  getAverageMark() {
+    return this.marks.reduce((a, b) => a + b) / this.marks.length;
+  }
+
+  getMinMark() {
+    return Math.min(...this.marks);
+  }
+
+  getMaxMark() {
+    return Math.max(...this.marks);
+  }
+}
+
+class Teacher extends Human {
+  constructor(info) {
+    super(info);
+    this.group = info.group;
+  }
+
+  getListOfNamesByAverageMark() {
+    return this.group
+      .sort((a, b) => b.getAverageMark() - a.getAverageMark())
+      .map((student) => student.getFullName());
+  }
+
+  getStudentByName(name) {
+    return this.group.find((student) => student.getFullName() === name);
+  }
+
+  removeStudentByName(name) {
+    this.group = this.group.filter((student) => student.getFullName() !== name);
+  }
+
+  updateStudentByName(student, name) {
+    for (let i = 0; i < this.group.length; i++) {
+      if (this.group[i].getFullName() === name) {
+        this.group[i] = student;
+        break;
+      }
+    }
```

Comment:
not bad, but there is a more declarative solution to this problem - try array.findIndex()
```javascript
const updatedStudentIndex = this.group.findIndex((student) => student.getFullName() === name);
this.group[updatedStudentIndex] = student;
```
just to let you know